pkg/queue/request_log.go

```diff
@@ -25,26 +25,47 @@ import (
 	"time"
 
 	pkghttp "github.com/knative/serving/pkg/http"
-	"github.com/knative/serving/pkg/utils"
 )
 
+type revInfo struct {
+	Name          string
+	Namespace     string
+	Service       string
+	Configuration string
+	PodName       string
+	PodIP         string
+}
+
+type respInfo struct {
+	Code    int
+	Size    int
+	Latency float64
+}
+
+type templateInput struct {
+	Request  *http.Request
+	Response *respInfo
+	Revision *revInfo
+}
+
 type requestLogHandler struct {
 	handler  http.Handler
 	writer   io.Writer
 	template *template.Template
+	revision *revInfo
 }
 
 // NewRequestLogHandler creates an http.Handler that logs request logs to an io.Writer.
-func NewRequestLogHandler(h http.Handler, w io.Writer, templateStr string) (http.Handler, error) {
+func NewRequestLogHandler(h http.Handler, w io.Writer, templateStr string,
+	ns, svc, config, rev, podName, podIP string) (http.Handler, error) {
```

Comment:
This is a ton of arguments.  Could we make the `revInfo` struct public and have the caller pass that to improve the clarity?

Reply:
Yup, will do.